packages/calcite-components/src/components/list/list.tsx

```diff
@@ -185,11 +185,13 @@ export class List implements InteractiveComponent, LoadableComponent {
     const target = event.target as HTMLCalciteListItemElement;
     const { listItems, selectionMode } = this;
 
-    listItems.forEach((listItem) => {
-      if (selectionMode === "single" || selectionMode === "single-persist") {
-        listItem.selected = listItem === target;
-      }
-    });
+    if (target.selected) {
```

Comment:
Should this also check if the selection mode is not `single` nor `single-persist`? When so, the `forEach` loop goes through all items without applying any changes.

Reply:
yeah nice catch 🎣